test/TestProjects/CadlFirstTest/Generated/cadl.json

```diff
@@ -0,0 +1,183 @@
+{
+  "Name": "demo.helloworld",
+  "Description": "",
+  "ApiVersions": ["0.1.0"],
+  "Clients": [
+    {
+      "Name": "HelloWorld",
+      "Operations": [
+        {
+          "Name": "topAction",
+          "Description": "",
+          "Parameters": [
+            {
+              "$id": "1",
+              "Name": "Endpoint",
+              "NameInRequest": "Endpoint",
+              "Description": "",
+              "Type": {"Name": "Uri", "Kind": "Uri"},
+              "Location": "Uri",
+              "IsRequired": true,
+              "IsEndpoint": true,
+              "Kind": "Client"
+            },
+            {
+              "Name": "action",
+              "NameInRequest": "action",
+              "Description": "",
+              "Type": {"Name": "string", "Kind": "String"},
+              "Location": "Path",
+              "IsRequired": true,
+              "IsApiVersion": false,
+              "IsResourceParameter": false,
+              "IsContentType": false,
+              "IsEndpoint": false,
+              "Kind": "Method",
+              "SkipUrlEncoding": true,
+              "Explode": false
+            },
+            {
+              "$id": "2",
+              "Name": "apiVersion",
+              "NameInRequest": "apiVersion",
+              "Description": "",
+              "Type": {"Name": "string", "Kind": "String"},
+              "Location": "Query",
+              "IsRequired": true,
+              "IsApiVersion": true,
+              "Kind": "Client"
+            }
+          ],
+          "Responses": [
+            {
+              "StatusCodes": [
+                200
+              ],
+              "BodyType": {"Name": "Thing", "Kind": "Object"},
+              "BodyMediaType": "Json"
+            }
+          ],
+          "HttpMethod": "GET",
+          "RequestBodyMediaType": "Json",
```

Comment:
Not right now. We don't support multipart yet, and it isn't clear if we are going to support it at all. We can change it in the future.

Reply:
I didn't get. We support it now. Why it is not supprted when we change to cadl?

Reply:
I remember we don't support multipart because right now we only support one body parameter, though in test server there are already test cases which have multiple body parameters.

Reply:
Maybe I misunderstood it. I though it is the [requestMedisTypes](https://github.com/Azure/autorest.csharp/blob/5cdef0c3582a80dce5b1b01abc112b8b661cfef2/test/TestServerProjects/media_types/Generated/CodeModel.yaml#L643). If it is not, could you point me to the counterpart in M4?

Reply:
RequestMediaTypes is a list:
![image](https://user-images.githubusercontent.com/8933858/180882582-2185d210-65f9-4e11-8e6a-0c07f53766d6.png)

Reply:
Thank you! What is counterpart of `RequestBodyMediaType` in swagger or M4?

Reply:
So, I saw you changed the name KnownMediaType to BodyMediaType. Why we have this change?

Reply:
To clarify that it is for the body. Implementation of the media types support is a bit messy here: convenience layer supports multiple media types, while protocol method has to handle everything in one place. And for these two scenarios, we pick one and the same information from different places. We'll merge it eventually, but I don't want to include it in this PR because it isn't a high priority